/*
    Una Proposal indica una richiesta di trasferimento di proprietà di un particolare batch da un azienda ad un'altra.
    La proposta è effettuata da una company ad un'altra tramite il Business Admin o un Operatore. 
    Il Business Admin o l'Operatore dell'azienda ricevente può accettare o rifiutare il passaggio. 
    Il Business Admin o l'operatore che inoltra la richiesta può cancellarla.
*/

syntax = "proto3";

message Proposal {

    // Stato della richiesta
    enum Status {
        OPEN = 0;
        ACCEPTED = 1;
        REJECTED = 2;
        CANCELED = 3;
    }

    // Id del batch da scambiare
    string batch_id = 1;

    // Timestamp di creazione della proposta
    uint64 timestamp = 2;

    // Identificativo della company mandante 
    string issuing_company = 3;

    // Identificativo della company ricevente 
    string receiving_company = 4;

    // Stato della proposta. (se accettata il batch passa alla receiving_company)
    Status status = 5;

    // Termini del trasferimento (una descrizione, breve)
    repeated string terms = 6;
}

message ProposalContainer {
    repeated Proposal entries = 1;
}